vllm/config.py

```diff
@@ -272,6 +275,163 @@ def is_init_field(cls: ConfigType, name: str) -> bool:
     return next(f for f in fields(cls) if f.name == name).init
 
 
+VerificationMethod = Literal["sigstore", "certificate", "key"]
+
+
+@config
+@dataclass(config=ConfigDict(arbitrary_types_allowed=True))
+class SignatureVerificationConfig:
+    """Signature verification configuration."""
+
+    verification_method: Optional[VerificationMethod] = None
+    """The signature verification method to use."""
+    signature: str = "model.sig"
+    """Path to the signature file. If only a filename is given, then a file
+    with this name is expected to hold the signature in the model's
+    directory."""
+    ignore_paths: Optional[Iterable[str]] = None
+    """File paths to ignore when verifying."""
+    ignore_git_paths: bool = True
+    """Ignore git-related files when verifying."""
+    identity: str = ""
+    """The email identity that signed with the 'sigstore' method."""
+    identity_provider: str = ""
+    """The email identity provider that was used when signing with the
+    'sigstore' method."""
+    use_staging: bool = False
+    """Use the Sigstore staging server."""
+    certificate_chain: Optional[str] = None
+    """The certificate chain needed for verifying with method 'certificate'."""
+    log_fingerprints: bool = False
+    """Whether to log certificate fingerprints when using the 'certificate'
+    method."""
+    public_key: Optional[str] = None
+    """Public key required for signature verification when using the 'key'
+    method."""
+    _verification_done: bool = False
+    """Set to True once signature verification was done."""
```

Comment:
Since this is considered state for this config, you could hint it as a [typing.ClassVar](https://docs.python.org/3/library/dataclasses.html#class-variables) so that it's not considered a `field` in the saame way as the other class members.

Might this be useful?

Reply:
> Might this be useful?

I don't know. I don't want it to be a class variable, but one per instance:

"When ClassVar is used, static type checkers will flag an error if an attempt is made to assign a value to that variable through an instance of the class. However, this restriction is not enforced at runtime by Python itself."

Reply:
Sounds good.